examples/parallel.rs

```diff
@@ -0,0 +1,351 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use cupcake::traits::*;
+use cupcake::DefaultShemeType;
+use cupcake::FVCiphertext;
+use cupcake::FVPlaintext;
+use cupcake::integer_arith::scalar::Scalar;
+use cupcake::SecretKey;
+
+use rayon::iter::{IntoParallelIterator,ParallelIterator,IndexedParallelIterator};
+
+use std::fmt::{ Display, Error, Formatter};
+
+pub struct ByteBuffer {
+    pub buffer: Vec<u8>,
+}
+
+impl ByteBuffer {
+    pub fn from_slice(v: &[u8]) -> ByteBuffer {
+        ByteBuffer { buffer: v.to_vec() }
+    }
+}
+
+impl Display for ByteBuffer {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+        //todo : fix me with proper formatting
+        for b in self.buffer.iter() {
+            write!(f, "{:X}", b)?;
+        }
+        Ok(())
+    }
+}
+
+fn smartprint<T: std::fmt::Debug>(v: &Vec<T>) {
+    println!("[{:?}, {:?}, ..., {:?}]", v[0], v[1], v[v.len() - 1]);
+}
+
+fn smartprint2<T: std::fmt::Debug>(v: &Vec< Vec<T>>) {
+    smartprint(&v[0]);
+    smartprint(&v[1]);
+    smartprint(&v[v.len() -1]);
+}
+
+
+// a struct for Parallel Cupcake operations
+
+    // usage for main method PS3I outline:
+    //      cargo run --example parallel
+    //      cargo run --release --example parallel
+    // usage for CupcakeParallel tests
+    //      cargo test --examples parallel
+    //      cargo test --release --examples parallel
+
+pub struct CupcakeParallel {
```

Comment:
Can we put the the struct and implementations in lib.rs? The goal is to keep example files strictly for examples.

Reply:
Also let's have some high level comments documenting the struct and its API